packages/utils/src/socialRemark/core.ts

```diff
@@ -0,0 +1,264 @@
+  SocialRemarkMessageAction,
+  SocialRemarkMessageProtocolName,
+  SocialRemarkMessageVersion,
+  SubSclSource
+} from './types'
+import { SocialRemarkConfig, SocialRemarkConfigData } from './config'
+import { decorateRemarkContentValue } from './decorators'
+
+export class SocialRemark {
+  private maybeRemarkMsg: unknown
+
+  /**
+   * Set global config for all new SocialRemark instances
+   * @param data: SocialRemarkConfigData
+   */
+  static setConfig(data: SocialRemarkConfigData) {
+    SocialRemarkConfig.getInstance().setConfig(data)
+  }
+
+  private msgParsed: SocialRemarkMessage<
+    SocialRemarkMessageAction,
+    boolean
+  > | null = null
+
+  private protNames: Set<SocialRemarkMessageProtocolName> = new Set(
+    SocialRemarkConfig.getInstance().config.protNames
+  )
+
+  private versions: Set<SocialRemarkMessageVersion> = new Set(
+    SocialRemarkConfig.getInstance().config.versions
+  )
+
+  private actions: Set<SocialRemarkMessageAction> = new Set(
+    SocialRemarkConfig.getInstance().config.actions
+  )
+
+  private msgDelimiter: string = '::'
+
+  /**
+   * Get SocialRemark full data, if it's available. Otherwise, go throw error.
+   */
+  public get message():
+    | SocialRemarkMessage<SocialRemarkMessageAction, boolean>
+    | never {
+    if (!this.msgParsed) throw new Error('Message is not available.')
+    return this.msgParsed!
+  }
+
+  /**
+   * Get SocialRemark instance message content, if it's valid and available.
+   */
+  public get content() {
+    return this.msgParsed && this.msgParsed.valid
+      ? this.msgParsed.content
+      : null
+  }
+
+  /**
+   * Get SocialRemark instance version, if it's valid and available.
+   */
+  public get version() {
+    return this.msgParsed ? this.msgParsed.version : null
+  }
+
+  /**
+   * Is SocialRemark valid.
+   */
+  public get isValidMessage(): boolean {
+    return !!this.msgParsed && this.msgParsed.valid
+  }
+
+  /**
+   * Convert bytes to string.
+   * @param src: unknown
+   */
+  static bytesToString(src: unknown): string {
+    if (!src || !Buffer.isBuffer(src)) return ''
+    return Buffer.from(src).toString('utf-8')
+  }
+
+  /**
+   * Create SocialRemark from raw string remark message.
+   * @param maybeRemarkMsg: unknown
+   */
+  public fromMessage(maybeRemarkMsg: unknown): SocialRemark {
+    this.maybeRemarkMsg = maybeRemarkMsg
+    this.parseMsg(maybeRemarkMsg)
+    return this
+  }
+
+  /**
+   * Create SocialRemark from source map.
+   * @param rmrkSrc: SubSclSource<SocialRemarkMessageAction>
+   */
+  public fromSource(
+    rmrkSrc: SubSclSource<SocialRemarkMessageAction>
+  ): SocialRemark {
+    this.isRemarkSourceValid(rmrkSrc)
+
+    try {
+      this.msgParsed = {
+        ...rmrkSrc,
+        valid: true
+      }
+      const contentPropsMap =
+        // @ts-ignore
+        REMARK_CONTENT_VERSION_ACTION_MAP[rmrkSrc.version][rmrkSrc.action]
+      for (const contentPropName in contentPropsMap) {
+        // @ts-ignore
```

Comment:
Instead of // @ts-ignore you can create const that cast type like this: `const contentPropName = key as ContentPropNameType`